osu.Game/Beatmaps/FramedBeatmapClock.cs

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Diagnostics;
+using osu.Framework;
+using osu.Framework.Allocation;
+using osu.Framework.Audio.Track;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Timing;
+using osu.Game.Configuration;
+using osu.Game.Database;
+using osu.Game.Screens.Play;
+
+namespace osu.Game.Beatmaps
+{
+    /// <summary>
+    /// A clock intended to be the single source-of-truth for beatmap timing.
+    ///
+    /// It provides some functionality:
+    ///  - Applies (and tracks changes of) beatmap, user, and platform offsets.
+    ///  - Adjusts <see cref="Seek"/> operations to account for said offsets, seeking in raw time values.
+    ///  - Exposes track length.
+    ///  - Allows changing the source to a new track (for cases like editor track updating).
+    /// </summary>
+    public class FramedBeatmapClock : Component, IFrameBasedClock, IAdjustableClock, ISourceChangeableClock
+    {
+        private readonly bool applyOffsets;
+
+        /// <summary>
+        /// The length of the underlying beatmap track. Will default to 60 seconds if unavailable.
+        /// </summary>
+        public double TrackLength => Track?.Length ?? 60000;
+
+        /// <summary>
+        /// The underlying beatmap track, if available.
+        /// </summary>
+        public Track? Track { get; private set; } // TODO: virtual rather than null?
+
+        /// <summary>
+        /// The total frequency adjustment from pause transforms. Should eventually be handled in a better way.
+        /// </summary>
+        public readonly BindableDouble ExternalPauseFrequencyAdjust = new BindableDouble(1);
```

Comment:
I was hoping this would come up as an issue. I'm honestly not sure why this works here. Testing against master, I also can't repro the test failures in #17302 (manually setting platform offset and then changing default/initial value to `1`). So something has changed to fix this in the interim.

I guess we could bisect back to see where this changed? Or just turn a blind eye for now?

The initial value of `1` as in this PR makes more sense to me.

Reply:
This is really weird, because I applied your experimental patch above onto `master` and got test failures back:

![image](https://user-images.githubusercontent.com/20418176/186020179-67111f92-f4cb-4e7d-9e7e-c3659a402919.png)

That said, these same tests pass on this branch, and I'm not really willing to spend time hunting for the reason why.